### PR TITLE
[WIP] to fix Issue #1

### PR DESCRIPTION
**Pull request by igaurab**

### CODE REVIEW

File: example/android/app/build.gradle [hunk 1]
```diff
@@ -39,7 +39,7 @@ android {
     defaultConfig {
         // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
         applicationId "np.com.naxa.graphhooper_route_example"
-        minSdkVersion 16
+        minSdkVersion 20
```
Comment (konishon) [inline example/android/app/build.gradle:42]: Why did you bump this?

File: android/src/main/kotlin/np/com/naxa/graphhooper_route/GraphhooperRoutePlugin.kt [hunk 2]
```diff
@@ -49,24 +70,70 @@ class GraphhooperRoutePlugin : MethodCallHandler {
...
+//            val request = DownloadManager.Request(Uri.parse(url))
+//            request.setAllowedNetworkTypes(DownloadManager.Request.NETWORK_MOBILE or DownloadManager.Request.NETWORK_WIFI)
+//            request.setTitle("Downloading $fileName")
+//            request.setDescription("Please wait")
+//            request.allowScanningByMediaScanner()
+//            request.setNotificationVisibility(DownloadManager.Request.VISIBILITY_VISIBLE_NOTIFY_COMPLETED)
+//            request.setDestinationInExternalPublicDir(Environment.DIRECTORY_DOWNLOADS,"nepal-gh")
+//
+//            val manager = ctx.applicationContext.getSystemService(Context.DOWNLOAD_SERVICE) as DownloadManager
+//            downloadId = manager.enqueue(request)
```
Comment (igaurab) [inline android/src/main/kotlin/np/com/naxa/graphhooper_route/GraphhooperRoutePlugin.kt:99]: Will implement these once we finalize where we will be hosting the file

File: android/src/main/kotlin/np/com/naxa/graphhooper_route/GraphhooperRoutePlugin.kt [hunk 3]
```diff
@@ -49,24 +70,70 @@ class GraphhooperRoutePlugin : MethodCallHandler {
...
+                    continue
+                val bufferedIs = BufferedInputStream(zip.getInputStream(entry))
+                bufferedIs.use {
+                    destFilePath.outputStream().buffered(1024).use { bos ->
+                        bufferedIs.copyTo(bos)
+                    }
+                }
+            }
+            Log.d("MPOINTS: ", mpoints.toString())
+            CalcPath().doInBackground(mpoints.get(0), mpoints.get(1), mpoints.get(2), mpoints.get(3))
```
Comment (igaurab) [inline android/src/main/kotlin/np/com/naxa/graphhooper_route/GraphhooperRoutePlugin.kt:130]: **Known Issue:**
Tried calling the function once the new map is unzipped in the Download. But returns error saying File not found.
After a hot reload everything seems to work

File: android/build.gradle [hunk 4]
```diff
@@ -48,5 +48,8 @@ dependencies {
         exclude group: 'org.apache.xmlgraphics', module: 'xmlgraphics-commons'
     }
     implementation 'com.google.code.gson:gson:2.8.5'
+    implementation 'androidx.localbroadcastmanager:localbroadcastmanager:1.0.0'
+    implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
+    implementation "org.jetbrains.kotlinx:kotlinx-serialization-runtime:0.13.0" // JVM dependency
```
Comment (igaurab) [inline android/build.gradle:53]: For converting Json to a Kotlin class, will remove these in next commit. [If not needed]

File: example/android/app/build.gradle (hunk 1 above)
Comment (igaurab) [reply example/android/app/build.gradle:42]: I think while I was trying to include Write Permissions in the manifest, there was error regarding the minSdk. Will review it.

File: android/src/main/kotlin/np/com/naxa/graphhooper_route/GraphhooperRoutePlugin.kt (hunk 2 above)
Comment (konishon) [reply android/src/main/kotlin/np/com/naxa/graphhooper_route/GraphhooperRoutePlugin.kt:99]: I have hosted it on firebase for now
https://firebasestorage.googleapis.com/v0/b/flutter-graphopper.appspot.com/o/nepal-gh.zip?alt=media&token=1026c9c3-b10f-43a2-b563-173c8ee556e6

maybe use this

**Pull request closed without merging**